src/tvheadend/HTSPVFS.cpp

```diff
@@ -108,15 +114,56 @@ ssize_t HTSPVFS::Read ( unsigned char *buf, unsigned int len )
   if (read > 0)
     m_offset += read;
 
+  /* Tvheadend may briefly return 0 bytes when playing an in-progress recording at end-of-file
+     we'll retry 50 times with 10ms pauses (~500ms) before giving up */
+  if (read == 0 && inprogress)
+  {
+    for (int i = 1; i <= 50; i++)
+    {
+      read = SendFileRead(buf, len);
+      if (read > 0)
+      {
+        m_offset += read;
+        break;
+      }
+      usleep(10000);
+    }
+  }
+
   return read;
 }
 
-long long HTSPVFS::Seek ( long long pos, int whence )
+long long HTSPVFS::Seek ( long long pos, int whence, bool inprogress )
 {
   if (m_fileId == 0)
     return -1;
 
-  return SendFileSeek(pos, whence);
+  long long ret = SendFileSeek(pos, whence);
+
+  /* for inprogress recordings see whether we need to toggle IsRealTimeStream */
+  if (inprogress)
+  {
+    int64_t fileSize = Size();
+    int64_t bitrate = 0;
+    m_eofOffsetSecs = -1;
+
+    if ((time(NULL) - m_fileStart) > 0)
+      bitrate = fileSize / (time(NULL) - m_fileStart);
```

Comment:
`time` call is rather expensive. So, just call it once and reuse the value. This is even more accurate.

And pleaes, use `nullptr` for new code (even if we still use `NULL ` at other places).

Reply:
Thanks for letting me know... I'm new to contributing to this and the kodi project, so picking up these things as I learn the ropes. Had only gone for time(NULL) as I could see that was how it was done elsewhere in this project.

Reply:
Done